### PR TITLE
fix(core): Make boolean config value parsing backward-compatible

### LINKED ISSUE #10543: Redis Server
### Bug Description

I created a redis database managed by Digital Ocean
The data below for connection.
N8n Cannot connect

username = default
password = ************************
host = db-redis-XXXXXXXXX-do-user-XXXXXX.f.db.ondigitalocean.com
port = 25061

LOG
2024-08-24T18:33:36.059Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"
2024-08-24T18:33:36.060Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"
2024-08-24T18:33:36.565Z | warn     | Error with Redis:  "{\n  errno: -32,\n  code: 'EPIPE',\n  syscall: 'write',\n  file: 'RedisServiceBaseClasses.js'\n}"
2024-08-24T18:33:36.566Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"
2024-08-24T18:33:36.567Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"
2024-08-24T18:33:37.071Z | warn     | Error with Redis:  "{\n  errno: -32,\n  code: 'EPIPE',\n  syscall: 'write',\n  file: 'RedisServiceBaseClasses.js'\n}"
2024-08-24T18:33:37.071Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"
2024-08-24T18:33:37.072Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"
2024-08-24T18:33:37.576Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"
2024-08-24T18:33:37.577Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"
2024-08-24T18:33:38.082Z | warn     | Error with Redis:  "{\n  errno: -104,\n  code: 'ECONNRESET',\n  syscall: 'write',\n  file: 'RedisServiceBaseClasses.js'\n}"
2024-08-24T18:33:38.083Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"
2024-08-24T18:33:38.083Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"
2024-08-24T18:33:38.588Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"
2024-08-24T18:33:38.588Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"
2024-08-24T18:33:39.092Z | warn     | Error with Redis:  "{\n  errno: -104,\n  code: 'ECONNRESET',\n  syscall: 'write',\n  file: 'RedisServiceBaseClasses.js'\n}"
2024-08-24T18:33:39.093Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"
2024-08-24T18:33:39.093Z | warn     | Redis unavailable - trying to reconnect... "{ file: 'redis-client.service.js' }"

### To Reproduce

      - EXECUTIONS_MODE=queue
      - QUEUE_BULL_REDIS_HOST=db-redis-XXXXXXXXX-do-user-XXXXXX.f.db.ondigitalocean.com
      - QUEUE_BULL_REDIS_PORT=25061
      - QUEUE_BULL_REDIS_USERNAME=default
      - QUEUE_BULL_REDIS_PASSWORD=**************
      - QUEUE_BULL_REDIS_TLS=TRUE

### Expected behavior

redis connection

### Operating System

Debian 12

### n8n Version

1.55.3

### Node.js Version

18.12.0

### Database

PostgreSQL

### Execution mode

queue

### PR DESCRIPTION
**Pull request by netroy**
## Summary
Sometimes people pass in `TRUE` or `1` instead on `true` for boolean values, and that seems to be parsed correctly by convict, but not by the new config format.
This PR fixes that parsing issue.

## Related Linear tickets, Github issues, and Community forum posts

Fixes https://github.com/n8n-io/n8n/issues/10543

## Review / Merge checklist

- [x] PR title and summary are descriptive
- [x] Tests included
- [ ] PR Labeled with `release/backport`

### CODE REVIEW

File: packages/@n8n/config/src/decorators.ts [hunk 1]
```diff
@@ -6,13 +6,25 @@ import { Container, Service } from 'typedi';
 type Class = Function;
 type Constructable<T = unknown> = new (rawValue: string) => T;
 type PropertyKey = string | symbol;
+type PropertyType = typeof Number | typeof Boolean | typeof String | Class;
```
Comment (tomi) [inline packages/@n8n/config/src/decorators.ts:9]: Is `typeof Number` somehow different from `number`? Or `Boolean` / `String`? Could we use the primitive ones instead?

File: packages/@n8n/config/src/decorators.ts [hunk 2]
```diff
@@ -6,13 +6,25 @@ import { Container, Service } from 'typedi';
...
-	type: unknown;
+	type: PropertyType;
 	envName?: string;
 }
 
 const globalMetadata = new Map<Class, Map<PropertyKey, PropertyMetadata>>();
 
+const readEnv = (envName: string) => {
+	const value = process.env[envName];
+	if (value) return value;
```
Comment (tomi) [inline packages/@n8n/config/src/decorators.ts:19]: This also ignores empty string `''`. Is that intended?

File: packages/@n8n/config/src/decorators.ts [hunk 3]
```diff
@@ -26,38 +38,28 @@ export const Config: ClassDecorator = (ConfigClass: Class) => {
...
+					if (isNaN(parsed)) {
+						console.warn(`Invalid number value for ${envName}: ${value}`);
+					} else {
+						config[key] = parsed;
 					}
 				} else if (type === Boolean) {
-					if (value !== 'true' && value !== 'false') {
-						// TODO: add a warning
-						value = undefined;
+					if (['true', 'TRUE', '1'].includes(value)) {
```
Comment (tomi) [inline packages/@n8n/config/src/decorators.ts:52]: What about `True`? Or `trUe`? Should the comparison be case-insensitive?

File: packages/@n8n/config/test/decorators.test.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+	it('should throw when explicit typing is missing', () => {
+		expect(() => {
+			@Config
+			class InvalidConfig {
+				@Env('STRING_VALUE')
+				value = 'string';
+			}
+			Container.get(InvalidConfig);
+		}).toThrowError('Invalid decorator metadata on key "value" on InvalidConfig');
```
Comment (tomi) [inline packages/@n8n/config/test/decorators.test.ts:17]: The thrown error message is not very actionable. What's the fix here? Could we include it in the error message

**Review by tomi — APPROVED** [review]
LGTM 🚀 Thank you for addressing the feedback 🙇

**Comment by janober** [comment]
Got released with [`n8n@1.57.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.57.0)

**Pull request merged**